### ISSUE TITLE
Probes fail to load on kernels that inline `do_renameat2`

### ISSUE THREAD

**Issue by rhysre** [elastic/ebpf#92]
We hook [do_renameat2](https://github.com/elastic/ebpf/blob/main/GPL/EventProbe/File/Probe.bpf.c#L235) to track renames. On AmazonLinux2 with a 5.10 kernel however, this function is inlined, causing our probes to fail to load. We may need to hook each rename syscall entry point instead.

**Comment by mmat11** [comment]
If I understood correctly, this is already fixed in `main`, can you confirm @rhysre ?

**Comment by lrishi** [comment]
This should also be fixed in 8.2, @rhysre did you have the latest commits when you tested this out? If not, please retest and close.

**Comment by rhysre** [comment]
Sorry for the late response, this got buried.

```
BPF check run at Mon Apr 11 21:34:26 UTC 2022

amazonlinux2 (14 kernel(s) to test)
[PASS] 5.10.102-99.473.amzn2.x86_64__0935484e5841bc500dea24692ee242c51513b5c8__06f8afb18db65cd79c2f51f767dfe
[PASS] 5.10.29-27.126.amzn2.x86_64__27df3384aff1a84c3b22bb814071e58f3ece3a0c__d771f9633d047683d45c99488b2904
[PASS] 5.10.29-27.128.amzn2.x86_64__750e9cfd8d1e105c452bc69a49ceaea9d1aec481__99983b608ea32959f6ad4954f0b6a3
[PASS] 5.10.35-31.135.amzn2.x86_64__f86fe5894f52a53213f5ed303f1a6a33b187d308__6ff61f75dc79c8ffad837ef8b13217
[PASS] 5.10.47-39.130.amzn2.x86_64__95eb2fcd20dd74fd68e8ede670d23990837421fb__313f15df7abce099b51048cba164c8
[PASS] 5.10.50-44.131.amzn2.x86_64__b2ea7f761beae2348781d709f4b6f1f9a66bdd46__32a06c62494515529666f98ddf2bd2
[PASS] 5.10.50-44.132.amzn2.x86_64__79c9f4a9cf36ba0af7692e9f52b26b50c5ec9fc1__018194457120233cc1980cc0cf3a11
[PASS] 5.10.59-52.142.amzn2.x86_64__c31c3a29a6699fdc5dce5fe96d95002e31dc1231__6b739f73347b29afc07691cd7a49a5
[PASS] 5.10.62-55.141.amzn2.x86_64__29b3e3bb8e935236486c5ad32eb068aa3758697b__8a2eda135e22111bbd45936d3890a5
[PASS] 5.10.68-62.173.amzn2.x86_64__045389e1cf65bb550c9cb0715d0f3703cb142066__bd980fc2aff66525c97b0b73dc106b
[PASS] 5.10.75-79.358.amzn2.x86_64__ed84f4433756b0be50e2df5f11c9ece47b6490bc__9dd164c6805c5007fae0d3431b1425
[PASS] 5.10.82-83.359.amzn2.x86_64__a4621859aacdfc67667d4119663571f9eb74617f__bd8c779129574004ccaac007b004f7
[PASS] 5.10.93-87.444.amzn2.x86_64__6500f29d401b1567e596b5d860746b9eadb681e1__053aa89e0f3f14e7d5634d47861b81
[PASS] 5.10.96-90.460.amzn2.x86_64__f28c9920ec2db19be4a00efa1efa38b7fbb0233c__dc0acdcb7de8114c7bd33a017a9903
```

👍 Yep, file access probes for the latest main successfully load and pass tests on all Amazon linux kernels as reported by the checker. This is fixed. Thanks @mmat11 .
